api/envoy/extensions/access_loggers/grpc/v3/als.proto

```diff
@@ -86,4 +87,23 @@ message CommonGrpcAccessLogConfig {
   // <envoy_v3_api_field_data.accesslog.v3.AccessLogCommon.filter_state_objects>`.
   // Logger will call `FilterState::Object::serializeAsProto` to serialize the filter state object.
   repeated string filter_state_objects_to_log = 5;
+
+  // Define the log condition as a filter to ensure that the destination is reached.
+  // Logs that match the filter are not forwarded to the normal endpoint,
+  // but are sent to a special endpoint. This endpoint will return ACK/NACK as a response,
+  // and Envoy will perform buffering upon receiving a NACK. The buffered message is
+  // resend until it is reached. Failure to send here means that
+  //
+  // 1. when a NACK message corresponding to the message sent is received
+  // 2. when the ACK message corresponding to the sent message buffer could not be
+  //    received until the message_ack_timeout is reached.
```

Comment:
What is the upper bound on these ACKed access logs? If the service is down for a long time, will the Envoy OOM?

Reply:
I added this issue as TODO but forgot to impl it ;)

Reply:
done

Reply:
Thanks. One nit is to ideally RST link to `message_ack_timeout`. I think though this is only a partial story; you're still going to indefinitely buffer, potentially all the way to OOM, if the access log service is down for a period of time, i.e. "the failed transmission is buffered again".